### DISCUSSION TITLE
Ensure Userland Proxy is Disabled

### DISCUSSION THREAD

**Discussion by alpana17** [cri-o/cri-o#8002] [Q&A]
The Docker engine provides two mechanisms for forwarding ports from the host to containers, hairpin NAT, and the use of a userland proxy. In most circumstances, the hairpin NAT mode is preferred as it improves performance and makes use of native Linux iptables functionality instead of using an additional component.
How does port forwarding work in CRI-O?

Do we need to disable something similar in CRI-O as well?

**Comment by alpana17** [comment]
@haircommander please suggest

**Comment by saschagrunert** [comment]
> How does port forwarding work in CRI-O?

We directly enter the network namespace of the pod and forward the data: 

https://github.com/cri-o/cri-o/blob/e4975d51365f2ec58465cc84a1700a076df5a7b1/internal/oci/runtime_oci_linux.go#L24-L97

> Do we need to disable something similar in CRI-O as well?

I don't think so.

**Reply by alpana17** [reply]
ok
Thank you for your inputs
